src/MapfileStyleParser.ts

```diff
@@ -373,25 +373,39 @@ export class MapfileStyleParser implements StyleParser {
       markSymbolizer.strokeWidth = mapfileStyle.width * 1;
     }
 
-    // TODO: Abstract to SVG and drop WellKnownName!
+    // Match points to WellKnownName
     const symbolType = mapfileStyle.symbol.type.toLowerCase();
-    switch (symbolType) {
-    case 'ellipse': {
-      const xy = mapfileStyle.symbol.points[0].split(' ');
-      if (xy[0] === xy[1]) {
-        markSymbolizer.wellKnownName = 'Circle' as WellKnownName;
+    const points = mapfileStyle.symbol.points.split(' ').map((item) => parseFloat(item));
+    if (symbolType === 'ellipse') {
+      if (!(points[0] === points[1] && points.length === 2)) {
+        console.warn('Custom Elipse not supported by MarkerSymbolyzer, fallback to \'Circle\'!');
+      }
+      markSymbolizer.wellKnownName = 'Circle' as WellKnownName;
+    } else {
+      switch (points) {
+      case [0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0, 0.0]:
```

Comment:
To have a better system, can you use a function here `assertSquare()` ?

Then something a litlle bit more more flexible, because we can start by another point than 0,0, or we can have a square with coordinates like 0-3 instead or 0-1: 

(eventually, add this in the  in the Useful.ts instead of in the big MapFileStyleParser.ts file)

```
private assertSquare(points: number[]): boolean {
  // Squares must have 5 coordinates
  if (points.length != 10) {
    return false;
  }
  // Squares must use only two differents number coordinates
  const firstCoordinate = points[0];
  const secondCoordinate = points.find(coordinate => coordinate !== firstCoordinate);
  if (points.find(coordinate => coordinate !== firstCoordinate && coordinate !== secondCoordinate) !== undefined) {
    return false;
  }

  const possibleSequence1 = [
    firstCoordinate, firstCoordinate,
    firstCoordinate, secondCoordinate,
    secondCoordinate, secondCoordinate,
    secondCoordinate, firstCoordinate,
    firstCoordinate, firstCoordinate,
  ];

  const possibleSequence2 = [
    firstCoordinate, firstCoordinate,
    secondCoordinate, firstCoordinate
    secondCoordinate, secondCoordinate,
   firstCoordinate, secondCoordinate,
    firstCoordinate, firstCoordinate,
  ];

  return assertSequence(possibleSequence1, points)) || assertSequence(possibleSequence2, points));
}
```

And the assertSequence method
```
export expectedSequence(expectedSequence: any[], points: any[]): boolean {
  const loopPoints = [...points, ...points];
  return loopPoints.join('-').includes(expectedSequence.join('-'));
}
```

It should works for all kind of squares without issue.
Same logic for the triangle and the cross.

If it make sens, please update your code.

Reply:
I tried to make it more generic in a bit different way. Somehow I don't get your approach in the example.